### PR TITLE
fixed error on person giving note

### PR DESCRIPTION
**Pull request by flutistar**
*Issue #:* /bcgov/entity#18373

*Description of changes:*
    * Set `hasNoPersonGivingNotice` flag based on `givingNoticeParty` value.
    * Set `initialAttention` value as empty string if the note is cancel note.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by cameron-eyds** [comment]
@flutistar When you want to generate a preview URL, comment `/gcbrun` 
This is basically a pre-viewable production build.

File: ppr-ui/src/composables/mhrInformation/useMhrUnitNote.ts [hunk 1]
```diff
@@ -190,6 +190,7 @@ export const useMhrUnitNote = () => {
     cancelUniNote.documentType = UnitNoteDocTypes.NOTE_CANCELLATION
     cancelUniNote.documentId = ''
     cancelUniNote.destroyed = false
+    cancelUniNote.hasNoPersonGivingNotice = note.givingNoticeParty ? false : true
```
Comment (cameron-eyds) [inline ppr-ui/src/composables/mhrInformation/useMhrUnitNote.ts]: Could be simplified if you prefer!
`cancelUniNote.hasNoPersonGivingNotice = !note.givingNoticeParty`

**Review by cameron-eyds — APPROVED** [review]
Looks good to me, just a small suggestion!

**Review by doug-lovett — APPROVED** [review]
Looks good.

**Comment by dimak1** [comment]
@flutistar I took a quick look and noticed few things in your PR:
- `package-lock.json` file is not updated to newer version, need to run `npm i`
- `isCancelUnitNote` requires `.value`

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue [hunk 2]
```diff
@@ -198,7 +198,7 @@ export default defineComponent({
 
     const localState = reactive({
       validateSubmittingParty: false,
-      initialAttention: getMhrUnitNoteRegistration.value?.attentionReference,
+      initialAttention: isCancelUnitNote ? '' : getMhrUnitNoteRegistration.value?.attentionReference,
```
Comment (cameron-eyds) [inline ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: As Dima mentioned in his review, you'll need to a property accessor for `.value`
(ie dot notation is pretty standard for us)

The reason why you wouldn't see an error for this, `isCancelUnitNote` is essentially just a permanent `true` boolean in the the context of this ternary(it's defined in the composable, so always true, not null or undefined etc).
Due to this, you wouldn't see a type error or anything in the console, it just always resolves to the first condition, ie ''. 

You will notice this in the UI however, where it seems to rely on the getter when navigating to and from the Review Confirm screen of a base unit note, to preserve the captured value (before submission). 

Give it a look and you will see what he means!

Comment (cameron-eyds) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: If your interested in more context on the **why** we need to do this, see vue's explanation of computed refs: https://vuejs.org/guide/essentials/computed.html

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1980-o1u77jfw.web.app

File: ppr-ui/src/components/unitNotes/UnitNoteReview.vue (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: Thanks for elaborating @cameron-eyds

Comment (flutistar) [reply ppr-ui/src/components/unitNotes/UnitNoteReview.vue]: Great!
I will fix them

**Review by dimak1 — APPROVED** [review]
Great!

**Review by cameron-eyds — APPROVED** [review]
LG!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1980-o1u77jfw.web.app

**Pull request merged**